nix/adafruit-nrfutil.nix

```diff
@@ -0,0 +1,31 @@
+{ fetchPypi
+, lib
+, python3Packages
+}:
+with python3Packages;
+buildPythonApplication rec {
+  pname = "adafruit-nrfutil";
+  version = "0.5.3.post16";
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-tyED3I5Q+SlR9RL8K28sYh2sFhRTHd+MC+OxPQIOJK0=";
+  };
+  build-system = [
+    nose
+    behave
+  ];
+  dependencies = [
+    pyserial
+    click
+    ecdsa
+  ];
+  # Tests were not updated to work on Python3, package does not support Python2.
+  doCheck = false;
+  meta = with lib; {
+    description = "Python 3 version of Nordic Semiconductor nrfutil utility and Python library (modified by Adafruit)";
+    # Nordic Semicondictor proprietary license
+    license = licenses.unfree;
```

Comment:
Can you source this for me, actually?

All of the files I see in <https://github.com/adafruit/Adafruit_nRF52_nrfutil> at a glance all look to have either BSD 3-clause or MIT

Reply:
Sure, it's in the python package metadata in [setup.py](https://github.com/adafruit/Adafruit_nRF52_nrfutil/blob/1361059009ff6a24d63b37eb3a4b28127837ead2/setup.py#L96)

And the header on [nordicsemi/dfu/signing.py](https://github.com/adafruit/Adafruit_nRF52_nrfutil/blob/1361059009ff6a24d63b37eb3a4b28127837ead2/nordicsemi/dfu/signing.py#L4-L5)

There's an open issue on the Adafruit_nRF52_nrfutil project about this license issue: https://github.com/adafruit/Adafruit_nRF52_nrfutil/issues/41

Reply:
Ah, that's unfortunate. I only spot checked a handful of files and that was not one of them

License here should probably be `[ licenses.bsd3 licenses.mit locenses.unfreeRedistributable ]` with a comment describing unfreeRedistributable and linking to that issue, I suppose?

Reply:
I've updated this line to be this list.

I originally copy-and pasted it, so the list included the typo `locenses` on `unfreeRedistributable`, which actually didn't throw an error, so it looks like `license` does not get evaluated properly when it's a list, as `unfreeRedistributable` should still throw the unfree issue.